src/components/SearchShopByName.vue

```diff
@@ -0,0 +1,49 @@
+<template>
+    <div class="search-by-name">
+        <input type="text" v-model="query" v-on:input="showSearchResults()">
+        <shop-list :shops="shops" />
+   </div>
+</template>
+
+<script>
+import { requestSearchByName } from '@/io/io';
+import ShopList from '@/components/ShopList';
+
+export default
+{
+    name: 'search-shop-by-name',
+    data ()
+    {
+        return {
+            query: '',
+            shops: [],
```

Comment:
So you suggest that the search results should be in the Store, so that they can be rendered onto a map?